guests/python/build.rs

```diff
@@ -11,13 +11,107 @@
 //! [CPython]: https://www.python.org/
 //! [Python Standard Library]: https://docs.python.org/3/library/index.html
 //! [`pyo3`]: https://pyo3.rs/
-use std::{fs::File, path::PathBuf};
+use sha2::Digest;
+use std::{
+    fs::File,
+    io::{Read, Write},
+    path::PathBuf,
+};
 
 fn main() {
+    println!("cargo:rerun-if-changed=build.rs");
+    download_wasi_sdk().expect("WASI SDK download");
     configure_linker();
     bundle_python_lib();
 }
 
+/// Download WASI SDK sysroot for static linking.
+///
+/// This downloads the WASI sysroot from the GitHub releases and extracts it to the downloads directory.
+/// If the sysroot already exists, this function returns early.
+fn download_wasi_sdk() -> Result<(), Box<dyn std::error::Error>> {
+    const WASI_SDK_VERSION_MAJOR: &str = "24";
+    const WASI_SDK_VERSION_MINOR: &str = "0";
+    const SHA256_WASI_SDK_SYSROOT: &str =
+        "35172f7d2799485b15a46b1d87f50a585d915ec662080f005d99153a50888f08";
+
```

Comment:
Reading through the interaction in the `Justfile` and your comment about the linker env var made me realize: you should check if this is a WASI build, otherwise you should skip the entire method. Two ways to do that from top of my head:

- pass some env file in `Justfile` as an indicator
- probably better: find some way to detect the current target from the build script, e.g. via `TARGET`, see https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts (i.e. `cargo should already set this for you, you just have to read it)